exercises/change/example.js

```diff
@@ -0,0 +1,139 @@
+class Change {}
+
+export default Change;
+
+// data structure to hold each candidate solution that is generated
+class Candidate {
+  constructor() {
+    this.wasSearched = false;
+    this.coins = [];
+  }
+
+  searched() {
+    this.wasSearched = true;
+  }
+
+  isSearched() {
+    return this.wasSearched;
+  }
+
+  getCoins() {
+    return this.coins;
+  }
+
+  addCoin(coin) {
+    const sortNum = (a, b) => a - b;
+
+    this.coins.push(coin);
+    this.coins.sort(sortNum);
+  }
+
+  getCoinCount() {
+    return this.coins.length;
+  }
+
+  getSum() {
+    const getSum = (total, num) => total + num;
+    return this.coins.reduce(getSum);
+  }
+}
+
+Change.prototype.calculate = (coinArray, target) => {
+  const candidates = [];
+  // fill the array with 0 to start
+  candidates[target] = 0;
+  candidates.fill(0);
+
+  // save a new candidate to the candidate array
+  const saveCandidate = (candidate) => {
+    const sum = candidate.getSum();
+
+    if (candidate.getSum() <= target &&
+            typeof (candidates[sum]) !== 'number' &&
+            candidates[sum].getCoinCount() > candidate.getCoinCount()) {
+      candidates[sum] = candidate;
+    }
+
+    if (candidate.getSum() <= target &&
+           typeof (candidates[sum]) === 'number') {
+      candidates[sum] = candidate;
+    }
+  };
+
+  // initialize the candidate array with the given coins only
+  const initialize = () => {
+    for (let j = 0; j < coinArray.length; j += 1) {
+      const temp = coinArray[j];
+      const candidate = new Candidate();
+      candidate.addCoin(temp);
+      saveCandidate(candidate);
+    }
+  };
+
+  // is everthing searched?
+  const isDone = () => {
+    let done = true;
+    for (let i = 0; i < candidates.length; i += 1) {
```

Comment:
Since you don't need the index `i` here, you might consider using the now available `for-of` instead, like so:

```javascript
for (let candidate of candidates){
...
}
```

This would allow you to eliminate the indirection of using the index variable `i` and the temporary variable `temp`.